packages/pwa-kit-dev/src/configs/webpack/overrides-plugin.js

```diff
@@ -37,8 +37,11 @@ class OverridesResolverPlugin {
 
         overridesFsRead.forEach((item) => {
             const end = item.substring(item.lastIndexOf('/index'))
-            const [l, ...rest] = item.split(/(index|\.)/)
-            this.extendsHashMap.set(l?.replace(overrideReplace, '').replace(/\/$/, ''), [end, rest])
+            const [l, ...rest] = item.split(/(index(?!(.test|.mock))|\.(?!(test|mock)))/)
```

Comment:
@vcua-mobify we actually want negative lookahead for any future `.` character something more along the lines of

`item.split(/(index\.(?!(\.))/)`

Reply:
Thanks for the input! I updated the regex to be the more generic 

`item.split(/(index(?!(\.[^\.]*\.))|\.(?!([^\.]*\.)))/)`

This splits on index only if there is 1 `.` after, or if not index this splits on the very last `.`